flac.md

```diff
@@ -630,6 +630,51 @@ Decoding the coded residual thus involves selecting the right coding method, fin
+
+Following the first packet are one or more header packets, each of which contains a single metadata block. The first of these packets MUST be a vorbis comment metadata block. This is contrary to unencapsulated FLAC streams, where the order of metadata blocks is not important except for the streaminfo block and where a vorbis comment metadata block is optional.
+
+Following the header packets are audio packets. Each audio packet contains a single FLAC frame. The first audio packet MUST start on a new Ogg page, i.e. the last metadata block MUST finish its page before any audio packets are encapsulated.
+
+The granule position of all pages containing header packets MUST be 0, for pages containing audio packets the granule position is the number of the last sample contained by the last completed packet in the frame. The sample numbering considers inter-channel samples. If a page contains no packet end (e.g. when page only contains the start of a large packet, which continues on the next page) then the granule position is set to the maximum value possible i.e. `0xFF 0xFF 0xFF 0xFF 0xFF 0xFF 0xFF 0xFF`.
+
+The granule position of the first audio data page with a completed packet MAY be larger than the number of samples contained in packets that complete on that page. In other words, the apparent sample number of the first sample in the stream following from the granule position and the audio data MAY be larger than 0. This allows for example a server to cast a live stream to several clients which joined at different moments, without rewriting the granule position for each client.
+
+In case an audio stream is encoded where these parameters change at some point in the stream, this should be dealt with by finishing encoding of the current Ogg stream and starting a new Ogg stream, concatenated to the previous one. This is called chaining in Ogg. See the Ogg specification for details.
+
+## Matroska mapping
+
+The Matroska container format is defined in [@!I-D.ietf-cellar-matroska]. The codec ID (EBML path `\Segment\Tracks\TrackEntry\CodecID`) assigned to signal tracks carrying FLAC data is `A_FLAC` in ASCII. All FLAC data before the first audio frame (i.e. the `fLaC` ASCII signature and all metadata blocks) is stored as CodecPrivate data (EBML path `\Segment\Tracks\TrackEntry\CodecPrivate`).
+
+Each FLAC frame (including all of its subframes) is treated as a single frame in the Matroska context.
+
+In case an audio stream is encoded where audio parameters (sample rate, number of channels or bit depth) change at some point in the stream, this should be dealt with a by finishing the current Matroska segment and starting a new one with the new parameters.
+
+## ISO Base Media File Format (MP4) mapping
+
+The encapsulation definition of FLAC audio in MP4 files was deemed too extensive to include in this document. A definition document can be found at https://github.com/xiph/flac/blob/master/doc/isoflac.txt
```

Comment:
I would prefer to have at least the definition of IDs and boxes, i.e. "Overview of Track Structure" and "FLACMetadataBlock"/"FLACSpecificBox" + list of authorized FLAC blocks, in addition to the link to the full details.

Reply:
I've added a summary, but I'm not knowledgeable enough to finish this. If anyone knows someone to contact to help out here, that would be much appreciated.

Otherwise I think a statement that audio parameters cannot change in a MP4 stream should probably be added, just like in the referenced document. I can't seem to find anything similar to Ogg's chaining and Matroska's segmenting.